engine/src/main/java/org/destinationsol/common/SolMath.java

```diff
@@ -464,18 +467,38 @@ private static float genLin(float b, float c) {
         return res < 0 ? Float.NaN : res;
     }
 
-    public static Vector2 readV2(JsonValue v, String name) {
+    /**
+     * Converts  string in given Json with given name to {@link Vector2}.
+     *
+     * The string has to be in form "0.5 0.5" for the returned {@code Vector2} to have values 0.5f, 0.5f.
+     * @param v Json where the string to be converted is located
+     * @param name Name of the string in given Json
+     * @return Converted {@code Vector2}, or null if the string was malformed
+     */
+    public static @Nullable Vector2 readV2(@NotNull JsonValue v, @NotNull String name) {
         return readV2(v.getString(name));
     }
 
-    public static Vector2 readV2(String encoded) {
+    /**
+     * Converts string in form "0.5 0.5" to {@link Vector2} with values 0.5f, 0.5f.
+     *
+     * @param encoded String in form of "0.5 0.5"
+     * @return Converted {@code Vector2}, or null if {@code encoded} was malformed
+     */
+    public static @Nullable Vector2 readV2(@NotNull String encoded) {
         String[] parts = encoded.split(" ");
-        float x = Float.parseFloat(parts[0]);
-        float y = Float.parseFloat(parts[1]);
-        return new Vector2(x, y);
+        Vector2 vector2;
+        try {
+            float x = Float.parseFloat(parts[0]);
+            float y = Float.parseFloat(parts[1]);
+            vector2 = new Vector2(x, y);
+        } catch (IndexOutOfBoundsException | NumberFormatException e) {
+            vector2 = null;
```

Comment:
By returning null, we are opening up the possiblity of NPE higher up. Either return Optional<Vector2> from here, or throw an exception.

Reply:
I'd prefer throwing an exception over returning `Optional.empty`. Too many optionals can also be bad :)

Reply:
Agreed.